contracts/bid/circuits/src/correctness.rs

```diff
@@ -0,0 +1,234 @@
+    /// Public input constructor, used when generating a proof.
+    pub pi_constructor: Option<Vec<PublicInput>>,
+}
+
+impl Circuit<'_> for CorrectnessCircuit {
+    fn gadget(
+        &mut self,
+        composer: &mut StandardComposer,
+    ) -> Result<Vec<PublicInput>, Error> {
+        let mut pi = vec![];
+
+        // Generate constant witness values for 0.
+        let zero =
+            composer.add_witness_to_circuit_description(BlsScalar::zero());
+
+        // Make sure we have all of the circuit inputs before proceeding.
+        let commitment = self
+            .commitment
+            .ok_or_else(|| CircuitErrors::CircuitInputsNotFound)?;
+        let value = self
+            .value
+            .ok_or_else(|| CircuitErrors::CircuitInputsNotFound)?;
+        let blinder = self
+            .blinder
+            .ok_or_else(|| CircuitErrors::CircuitInputsNotFound)?;
+
+        // Allocate all private inputs to the circuit.
+        let value = AllocatedScalar::allocate(composer, value);
+        let blinder = AllocatedScalar::allocate(composer, blinder);
+
+        // ------------------------------------------------------- //
+        //                                                         //
+        //                   Correctness circuit                   //
+        //                                                         //
+        // ------------------------------------------------------- //
+
+        // 1. Prove knowledge of commitment pre-image.
+        let p1 = scalar_mul(composer, value.var, GENERATOR_EXTENDED);
+        let p2 = scalar_mul(composer, blinder.var, GENERATOR_NUMS_EXTENDED);
+        let computed_commitment = p1.point().fast_add(composer, *p2.point());
+
+        // Add PI constraint for the commitment computation check.
+        pi.push(PublicInput::AffinePoint(
+            commitment,
+            composer.circuit_size(),
+            composer.circuit_size() + 1,
+        ));
+
+        // Ensure equality between the computed commitment and the provided one.
+        composer.assert_equal_public_point(computed_commitment, commitment);
+
+        // 2. Range check - v_min <= value <= v_max
+        let cond = range_check(
+            composer,
+            BlsScalar::from(*V_MIN),
+            BlsScalar::from(*V_MAX),
+            value,
+        );
+        // We need cond to be one, so we conditionally select one.
+        let cond = conditionally_select_one(composer, zero, cond);
+        // Constrain cond to be one - meaning that the range check holds.
+        composer.constrain_to_constant(
+            cond,
+            BlsScalar::one(),
+            BlsScalar::zero(),
+        );
+
+        self.size = composer.circuit_size();
+        Ok(pi)
+    }
+
+    fn compile(
+        &mut self,
+        pub_params: &PublicParameters,
+    ) -> Result<(ProverKey, VerifierKey, usize), Error> {
+        // Setup PublicParams
+        let (ck, _) = pub_params.trim(1 << 16)?;
```

Comment:
The circut is probably smaller. You should try to get the lowest value possible.